File: Data Structures & Algorithms/is-palindrome/submission-1.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0;
        int right = s.length() - 1;

        while(left < right) {
            while(left < right && !isalnum(s[left])) {
                left++;
            }
            while(right > left && !isalnum(s[right])) {
                right--;
            }

            transform(s.begin(), s.end(), s.begin(), ::tolower);

            if(s[left] != s[right]) {
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        if (prices.length < 2) return 0
        let maxProfit = 0;

        let left = 0;
        let right = left + 1;

        while (left < right) {
            const currMax = prices[right] - prices[left];
            if (currMax > maxProfit) {
                maxProfit = currMax;
            }

            if (right === prices.length - 1) {
                if (left === prices.length - 2) {
                    return maxProfit;
                }

                left++;
                right = left + 1;
            } else {
                right++;
            }
        }

        return maxProfit;
    }
}
